tests/integration_tests/data_steward/validation/achilles_heel_test.py

```diff
@@ -64,18 +63,21 @@ def _load_dataset(self, hpo_id):
                 dataset=self.dataset, vocab=common.VOCABULARY_DATASET)
             bq_utils.query(q)
 
-    @staticmethod
-    def get_mock_hpo_bucket():
+    def get_mock_hpo_bucket(self):
         bucket_env = 'BUCKET_NAME_' + FAKE_HPO_ID.upper()
         hpo_bucket_name = os.getenv(bucket_env)
-        if hpo_bucket_name is None:
+        if not hpo_bucket_name:
             raise EnvironmentError()
-        return hpo_bucket_name
+        return self.storage_client.get_hpo_bucket(hpo_bucket_name)
 
-    @mock.patch('gcs_utils.get_hpo_bucket')
+    @mock.patch.object(StorageClient, 'get_hpo_bucket')
     def test_heel_analyses(self, mock_hpo_bucket):
         # Long-running test
-        mock_hpo_bucket.return_value = self.get_mock_hpo_bucket()
+        mock_bucket = mock.MagicMock()
+        mock_hpo_bucket.return_value = mock_bucket
+
+        mock_hpo_site: str = self.get_mock_hpo_bucket()
+        type(mock_bucket).name = mock.PropertyMock(return_value=mock_hpo_site)
```

Comment:
Interesting, is this mocking the type of mock_bucket to return as mock_hpo_site?

Reply:
it's an interesting pattern.  It adds the `name` attribute.  When code needs the attribute (`bucket.name`) then the returned value is simply just that.  I can't quite understand how this syntax works, but it does.

Reply:
I see, it's adding the attribute. Thanks